File: Cpp/832_Flipping_an_Image.cpp
class Solution {
public:
    vector<vector<int>> flipAndInvertImage(vector<vector<int>>& A) {
        int m = A.size();
        int n = A[0].size();
        vector<vector<int>> result(m, vector<int>(n));
        for(int i=0; i<m; ++i){
            for(int j=n-1; j>=0; --j){
                result[i][n-j-1] = A[i][j] ^ 1;
            }
        }
        
        return result;
    }
};

File: Kotlin/832_Flipping_an_Image.kt
class Solution {
    fun flipAndInvertImage(image: Array<IntArray>): Array<IntArray> {
        val m = image.size
        val n = image[0].size
        val result = Array(m) { IntArray(n) }
        
        for(i in image.indices) {
            for(j in image[i].size - 1 downTo 0) {
                result[i][image[i].size - 1 - j] = image[i][j] xor 1
            }
        }
        
        return result
    }
}